DataFormats/TrackerCommon/interface/TrackerTopology.h

```diff
@@ -600,6 +600,13 @@ class TrackerTopology {
   bool hasField(const DetId &id, DetIdFields idx) const {
     return id.subdetId() == bits_per_field[idx].subdet;
   }
+
+  const PixelBarrelValues& pbVals() const { return pbVals_; }
+  const PixelEndcapValues& pfVals() const { return pfVals_; }
+  const TOBValues& tobVals() const { return tobVals_; }
+  const TIBValues& tibVals() const { return tibVals_; }
+  const TIDValues& tidVals() const { return tidVals_; }
+  const TECValues& tecVals() const { return tecVals_; }
```

Comment:
i gather from the comment that you did - seems like the size penalty would be small and avoids duplicating all the parsing of detid code and internals. (squashing the various fields into one set of fields based on subdetid seems not hard)